app/views/transactions/_form.html.erb

```diff
@@ -1,26 +1,29 @@
-<%= form_with model: transaction do |form| %>
+<div class="flex flex-col justify-center p-4" data-controller="transaction-form">
 
-  <div class="flex flex-col justify-center p-4">
-    <%#= form.label :symbol, "Search Stock Symbol" %>
-    <%#= form.text_field :symbol, class: "form-control" %>
-
-    <%= render_filter items, class: "p-1" do %>
-      <%= filter_icon do %>
-        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="mr-2 h-4 w-4 shrink-0 opacity-50 ml-2"><circle cx="11" cy="11" r="8"></circle><line x1="21" x2="16.65" y1="21" y2="16.65"></line></svg>
-      <% end %>
+  <%= render_filter items, placeholder: "Search Stock", data: { action: "click->transaction-form#updateSymbol"}, class: "p-1" do %>
+    <%= filter_icon do %>
+      <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="mr-2 h-4 w-4 shrink-0 opacity-50 ml-2"><circle cx="11" cy="11" r="8"></circle><line x1="21" x2="16.65" y1="21" y2="16.65"></line></svg>
     <% end %>
-    
-    <%= form.label :price, "Price" %>
-    <%= form.text_field :price %>
-    
-    <%= form.label :currency, "Currency" %>
-    <%= form.text_field :currency %>
+  <% end %>
+
+  <%= form_with model: transaction do |form| %>
```

Comment:
form inputs are missing autocomplete attributes. Are there warnings in the browser?

Reply:
let's make price and quantity number fields